support/mender-inventory-provides

```diff
@@ -10,6 +10,25 @@ set -e
 # Also submit device_type, which is in a separate file, not in the database. We don't know if the
 # file contains a newline, so it's important that this is done last, to avoid corrupting other
 # entries.
-grep '^device_type=' /var/lib/mender/device_type
+
+# Set default value for DeviceTypeFile
+default_device_type_file="/var/lib/mender/device_type"
+
+# Read mender.conf file to find the DeviceTypeFile key if it exists prioritizing /etc over /var/lib
+if grep -q '"DeviceTypeFile"' /etc/mender/mender.conf; then
```

Comment:
`/etc/mender/mender.conf`
This can be influenced by env variables and flags

```
mender-update -h
NAME:
   mender-update - manage and start Mender Update
...
...
Environment variables:
   - MENDER_CONF_DIR - configuration (default: /etc/mender).
   - MENDER_DATA_DIR - identity, inventory and update modules (default: /usr/share/mender).
   - MENDER_DATASTORE_DIR - runtime datastore (default: /var/lib/mender).

COMMANDS:
...
GLOBAL OPTIONS:
   --config FILE, -c FILE           Configuration FILE path (default:
                                    /etc/mender/mender.conf)
   --fallback-config FILE, -b FILE  Fallback configuration FILE path (default:
                                    /var/lib/mender/mender.conf)
```

Don't have an idea on top of my head on how to approach this.
Just a note that the the hardcoded `mender.conf` may become the new `/var/lib/mender/device_type` :)

Reply:
I assumed that despite you change the env variables, the /etc/mender/mender.cond and /var/lib/mender/mender.conf will not. I know it doesn't make that much sense, but otherwise we need to also alter the first three paragraphs as part of a parallel PR: https://docs.mender.io/client-installation/configuration#configuration

What do you think @TheMeaningfulEngineer ?

Reply:
@TheMeaningfulEngineer what do you think about bff1a4ff09dd646023fe10b3dbf08df939aaff87? 
I don't know how to handle if mender-update was executed with any of the follow ones:

```
--config FILE, -c FILE 
--fallback-config FILE, -b FILE
--data DIR, -d DIR
```

Any advise @lluiscampos?

Reply:
We cannot possibly get the command line options of mender-auth or mender-update from the script. It is correct to use the environment variables (good call on that too!).

In a way the more we align this code with [the rootfs-image](https://github.com/mendersoftware/mender/blob/4.0.5/support/modules/rootfs-image#L44) the better.